GameOfRunes/GameOfRunes/GameEngine/ContactDelegate.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  ContactDelegate.swift
+//  GameOfRunes
+//
+//  Created by Andy on 26/3/20.
+//  Copyright © 2020 TeamHoWan. All rights reserved.
+//
+
+import SpriteKit
+import GameplayKit
+
+class ContactDelegate: NSObject, SKPhysicsContactDelegate {
+    private weak var gameEngine: GameEngine?
+    
+    init(gameEngine: GameEngine) {
+        self.gameEngine = gameEngine
+        
+    }
+    
+    func didBegin(_ contact: SKPhysicsContact) {
+        let isBodyAEnemy = contact.bodyA.categoryBitMask == CollisionType.enemy.rawValue
+        let isBodyAEndPoint = contact.bodyA.categoryBitMask == CollisionType.endpoint.rawValue
+        let isBodyAPowerUp = contact.bodyA.categoryBitMask == CollisionType.powerUp.rawValue
+        let isBodyBEnemy = contact.bodyB.categoryBitMask == CollisionType.enemy.rawValue
+        let isBodyBEndPoint = contact.bodyB.categoryBitMask == CollisionType.endpoint.rawValue
+        let isBodyBPowerUp = contact.bodyB.categoryBitMask == CollisionType.powerUp.rawValue
+    
+        guard let nodeA = contact.bodyA.node as? CollisionNode,
+            let nodeB = contact.bodyB.node as? CollisionNode else {
+                return
+        }
+        
+        if isBodyAEnemy {
```

Comment:
NIT: maybe this section can be re-factored out into a separate private function. In the future, if another body type is added, then you would be able to isolate the change to that function. Right now, there's only 3 body types so i guess it's more manageable :)